src/main/java/com/dabsquared/gitlabjenkins/publisher/GitLabMessagePublisher.java

```diff
@@ -35,6 +36,10 @@
     private String abortNoteText;
     private String unstableNoteText;
 
+    /**
+     * @deprecated use {@link #GitLabMessagePublisher()} with setters to configure an instance of this class.
+     */
+    @Deprecated
     @DataBoundConstructor
```

Comment:
Should this `@DataBoundConstructor` be around the `public GitLabMessagePublisher() { }` constructor instead?

Reply:
I must admit I'm somewhat unsure exactly where `@DataBoundConstructor` is actually required now that `@DataBoundSetter` has been added. 

When making these changes I used [this](https://github.com/jenkinsci/slack-plugin/pull/236/files) merge from the `slack-plugin` as a reference. Looking again at those changes they do appear to have moved `@DataBoundConstructor` annotation onto the no-arg constructors. As such I will make the same change.